azurerm/resource_arm_container_group_test.go

```diff
@@ -81,6 +83,9 @@ func TestAccAzureRMContainerGroup_imageRegistryCredentialsUpdate(t *testing.T) {
 					resource.TestCheckResourceAttr(resourceName, "image_registry_credential.0.server", "hub.docker.com"),
 					resource.TestCheckResourceAttr(resourceName, "image_registry_credential.0.username", "updatedusername"),
 					resource.TestCheckResourceAttr(resourceName, "image_registry_credential.0.password", "updatedpassword"),
+					resource.TestCheckResourceAttr(resourceName, "container.0.ports.#", "1"),
+					resource.TestCheckResourceAttr(resourceName, "container.0.ports.0.port", "80"),
```

Comment:
This will fail because with a set you can no longer use `.0.` notation.

Reply:
So what can I do?

Reply:
The best way is to simply check the count via #. You can calculate the hash and access it that way however that isn't required.